releases/release-4.0.9.md

```diff
@@ -0,0 +1,103 @@
+    - 添加了对于连接的空闲时间的监控项 [#21301](https://github.com/pingcap/tidb/pull/21301)
+
++ TiKV
+
+    -
+
++ PD
+
+    -
+
++ TiFlash
+
+    - 降低 Replica read 时的延迟
+    - 优化 TiFlash 的错误信息
+    - 优化在大数据量下，对缓存数据大小的限制
+    - 添加正在处理的 coprocessor 请求数量的 metrics
+
++ Tools
+
+    -
+
+## Bug 修复
+
++ TiDB
+
+    - 修复了前缀索引和 `OR` 条件一起使用时结果不正确的问题 [#21287](https://github.com/pingcap/tidb/pull/21287)
+    - 修复了开启自动重试后可能出现的一处 panic。在事务重试时，执行到 `SET SQL_SELECT_LIMIT=DEFAULT` 这类语句时会触发该 panic [#21285](https://github.com/pingcap/tidb/pull/21285)
+    - 修复了分区表对于列的类型检查的一处问题。对于时间和日期类型，错误地使用了字符串方式比较大小 [#21273](https://github.com/pingcap/tidb/pull/21273)
+    - 修复了分区表对于列的类型检查的一处问题。分区表达式的值的类型和分区列的类型必须一致 [#21136](https://github.com/pingcap/tidb/pull/21136)
+    - 修复了哈希分区表对于分区名唯一性检查的问题 [#21257](https://github.com/pingcap/tidb/pull/21257)
+    - 修复非 int 类型的值，插入到哈希分区表后结果不正确的问题 [#21238](https://github.com/pingcap/tidb/pull/21238)
+    - 修复了在部分写入类的场景中，使用了 index join 会遇到非预期报错的问题 [#21249](https://github.com/pingcap/tidb/pull/21249)
+    - 修复了在 CASE WHEN 中 bigint 无符列的值，被错误地转换成有符类型的问题 [#21236](https://github.com/pingcap/tidb/pull/21236)
+    - 修复了 index hash join 和 index merge join 没有考虑 collation 的问题 [#21219](https://github.com/pingcap/tidb/pull/21219)
+    - 修复了分区表在建表和查询时，没有考虑 collation 的问题 [#21181](https://github.com/pingcap/tidb/pull/21181)
+    - 修复了慢日志记录的查询结果可能不全的问题 [#21211](https://github.com/pingcap/tidb/pull/21211)
+    - 修复了一处数据库名大小写处理不当，导致的 `DELETE` 未正确删除数据的问题 [#21206](https://github.com/pingcap/tidb/pull/21206)
+    - 修复了一处执行 DML 语句导致 schema 的内存被覆盖的问题 [#21050](https://github.com/pingcap/tidb/pull/21050)
+    - 修复了使用 join 时，无法查询到合并后的列的问题 [#21021](https://github.com/pingcap/tidb/pull/21021)
+    - 修复了一些 semi join 的查询结果不正确的问题 [#21019](https://github.com/pingcap/tidb/pull/21019)
+    - 修复了表锁对于 update 语句不生效的问题 [#21002](https://github.com/pingcap/tidb/pull/21002)
+    - 修复创建递归的视图，出现栈溢出的问题 [#21001](https://github.com/pingcap/tidb/pull/21001)
+    - 修复了 index merge join 在执行外连接的时候，结果不符合预期的问题 [#20954](https://github.com/pingcap/tidb/pull/20954)
+    - 修复了一处事务问题，该场景下应该返回结果未知，但是却返回了执行失败 [#20925](https://github.com/pingcap/tidb/pull/20925)
+    - 修复 `explain for connection` 无法显示最后一次执行计划的问题 [#21315](https://github.com/pingcap/tidb/pull/21315)
+    - 修复在 RC 隔离级别下，index merge 结果不正确的问题 [#21253](https://github.com/pingcap/tidb/pull/21253)
+    - 修复了由于事务写冲突重试导致的 auto-id 分配失败 [#21079](https://github.com/pingcap/tidb/pull/21079)
+    - 修复了 JSON 数据无法通过 `load data` 无法正确导入到 TiDB 的问题 [#21074](https://github.com/pingcap/tidb/pull/21074)
+    - 修复新增加 enum 类型的列的默认值问题 [#20998](https://github.com/pingcap/tidb/pull/20998)
+    - 对于日期类型的数学计算，保留原始的数据类型信息，修复 `adddata` 函数插入非法值的问题 [#21176](https://github.com/pingcap/tidb/pull/21176)
+    - 修复了部分场景错误地生成了 PointGet 的执行计划，导致执行结果不正确 [#21244](https://github.com/pingcap/tidb/pull/21244)
+
++ TiKV
+
+    -
+
++ PD
+
+    -
+
++ TiFlash
+
+    - 修复 `INFORMATION_SCHEMA.CLUSTER_HARDWARE` 中可能包含未被使用的硬盘信息的问题
+    - 修复 Delta cache 内存占用量估算偏少的问题
+
++ Tools
+
+    -
```

Comment:
```suggestion
+ Tools

# TiCDC

## 优化提升

* 添加对 TiKV 开启 hibernate region 的告警规则 [#1120](https://github.com/pingcap/ticdc/pull/1120)
* 优化 schema storage 的内存使用 [#1127](https://github.com/pingcap/ticdc/pull/1127)

## Bug 修复

* 修复某些异常情况下存在多个 Owner 的问题 [#1104](https://github.com/pingcap/ticdc/pull/1104)

# TiDB-binlog

## 优化提升

* 在写下游出错时给 drainer 设置非零退出码 [#1012](https://github.com/pingcap/tidb-binlog/pull/1012)

# BR

## 优化提升

* BR 不再接受存在歧义的 `--checksum false` 命令行参数，正确用法为 `--checksum=false` [#588](https://github.com/pingcap/br/pull/588)
* 支持暂时性地调整 PD 的参数 [#596](https://github.com/pingcap/br/pull/596)
* 支持恢复数据表的统计信息 [#622](https://github.com/pingcap/br/pull/622)
* 对于可恢复错误 `read index not ready` 和 `proposal in merging mode` 进行重试 [#626](https://github.com/pingcap/br/pull/626)

## Bug 修复

* 修复因 S3 secret access keys 中存在特殊字符而导致失败的问题 [#617](https://github.com/pingcap/br/pull/617)

# Dumpling

## 优化提升

* 对导出失败部分的数据进行重试 [#182](https://github.com/pingcap/dumpling/pull/182)
* 支持同时设置 `-F` 和 `-r` 两个参数 [#177](https://github.com/pingcap/dumpling/pull/177)
* 默认不导出系统表 [#194](https://github.com/pingcap/dumpling/pull/194)
* 在设置 `--transactional-consistency` 参数时支持重建 MySQL 链接 [#199](https://github.com/pingcap/dumpling/pull/199)

## Bug 修复

* 修复在某些情况下 MySQL 链接关闭导致 Dumpling 卡住的问题 [#190](https://github.com/pingcap/dumpling/pull/190)

# TiDB-Lightning

## 优化提升

* 默认不恢复系统表 [#459](https://github.com/pingcap/tidb-lightning/pull/459)
* 支持为 auto random 的主键设置默认值 [#457](https://github.com/pingcap/tidb-lightning/pull/457)
* 完善 local 模式下分裂 region 的精度 [#422](https://github.com/pingcap/tidb-lightning/pull/422)
* 支持给 `tikv-importer.region-split-size`, `mydumper.read-block-size`, `mydumper.batch-size` and `mydumper.max-region-size` 设置可读的参数，比如 "2.5 GiB" [#471](https://github.com/pingcap/tidb-lightning/pull/471)

## Bug 修复

* 修复使用错误信息编码 key 的问题 [#437](https://github.com/pingcap/tidb-lightning/pull/437)
* 修复 GC life time ttl 不生效的问题 [#448](https://github.com/pingcap/tidb-lightning/pull/448)
* 修复手动关闭时可能出现的 panic 问题 [#484](https://github.com/pingcap/tidb-lightning/pull/484)

```

Reply:
added in https://github.com/pingcap/docs-cn/pull/4976/commits/1966d63d5e928aaad3277e23aea87c5cc8aedad6.